src/ResourceGenerator/ExtractCollectionTrait.php

```diff
@@ -37,11 +37,14 @@ trait ExtractCollectionTrait
         ServerRequestInterface $request
     ) : Link;
 
+    /**
+     * @return Link
```

Comment:
Well, removing a type-hint is a BC break and thus needs a major version bump.

Is it possible to throw the unused method "away" by using this work around:

```php
use ExtractCollectionTrait {
    generateSelfLink as unusedGenerateSelfLink;
}
```

This method is 100% overridden by the implementing class and thus should work.

I was not even aware that methods from traits are not imported when a method already exists, I've thought one has to explicitly use the `use` statement to rename a duplicated method from within a trait...

PHP always gives me bad feeling...

Reply:
Thanks for your suggestion. indeed removing it in the trait also gave me a feeling it could also be a breaking change, especially since userland code could also be using the trait directly.

Although your suggestion didn't work out of the box, it hinted me about a possible solution by creating a specific trait with the same signature as the package's implementation, and overriding it.